lib/Target/TfheRustBool/TfheRustBoolEmitter.cpp

```diff
@@ -165,29 +168,70 @@ void TfheRustBoolEmitter::emitAssignPrefix(Value result) {
   os << "let " << variableNames->getNameForValue(result) << " = ";
 }
 
+void TfheRustBoolEmitter::emitReferenceConversion(Value value) {
```

Comment:
nit: could you also add a quick assertion here that the value element type is a EncryptedUint (or maybe add this to the doc-string in the header file that it's expected to run on shaped types of encrypted values?)

Reply:
The tfhe-rs-bool emitter only produces EncryptedBools
I added an additional check